Tests/ObjectMapperTests/MapContextTests.swift

```diff
@@ -70,6 +72,179 @@ class MapContextTests: XCTestCase {
 		XCTAssertNil(person?.name)
 	}
 	
+	// MARK: Nested
+	func testNestedMappingWithContext() {
+		let JSON = ["person": ["name": "Tristan"]]
+		let context = Context(shouldMap: true)
+		
+		let nestedPerson = Mapper<NestedPerson>(context: context).map(JSON: JSON)
+		
+		XCTAssertNotNil(nestedPerson)
+		XCTAssertNotNil(nestedPerson?.person?.name)
+	}
+	
+	func testNestedMappingWithContextViaMappableExtension() {
+		let JSON = ["person": ["name": "Tristan"]]
+		let context = Context(shouldMap: true)
+		
+		let nestedPerson = NestedPerson(JSON: JSON, context: context)
+		
+		XCTAssertNotNil(nestedPerson)
+		XCTAssertNotNil(nestedPerson?.person?.name)
+	}
+
+	func testNestedMappingWithoutContext() {
+		let JSON = ["person": ["name": "Tristan"]]
+		
+		let nestedPerson = Mapper<NestedPerson>().map(JSON: JSON)
+		
+		XCTAssertNotNil(nestedPerson)
+		XCTAssertNil(nestedPerson?.person?.name)
+	}
+	
+	// MARK: Array
+	func testArrayMappingWithContext() {
+		let JSON = ["persons": [["name": "Tristan"], ["name": "Anton"]]]
+		let context = Context(shouldMap: true)
+		
+		let person = Mapper<PersonList>(context: context).map(JSON: JSON)
+		
+		XCTAssertNotNil(person)
+		XCTAssertNotNil(person?.persons)
+	}
+	
+	func testArrayMappingWithContextViaMappableExtension() {
+		let JSON = ["persons": [["name": "Tristan"], ["name": "Anton"]]]
+		let context = Context(shouldMap: true)
+		
+		let person = PersonList(JSON: JSON, context: context)
+		
+		XCTAssertNotNil(person)
+		XCTAssertNotNil(person?.persons)
+	}
+	
+	func testArrayMappingWithoutContext() {
+		let JSON = ["persons": [["name": "Tristan"], ["name": "Anton"]]]
+		
+		let person = Mapper<PersonList>().map(JSON: JSON)
+		
+		XCTAssertNotNil(person)
+		XCTAssertNil(person?.persons)
+	}
+	
+	// MARK: ImmutableMappable
+	// MARK: Single
+	func testImmatableMappingWithContext() {
+		let JSON = ["name": "Anton"]
+		let context = ImmutableContext(isDeveloper: true)
+		
+		let person = try? Mapper<ImmutablePerson>(context: context).map(JSON: JSON)
+		
+		XCTAssertNotNil(person)
+		
+		XCTAssertEqual(person?.isDeveloper ?? !context.isDeveloper, context.isDeveloper)
```

Comment:
Why did you use `?? !context.isDeveloper` here? If it was just for making `Bool?` to `Bool`, wouldn't it better using `?? false` or `person!.isDeveloper`? (we have ensured that the `person` is not nil from the test above)